File: LeetCode/length_of_last_word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int len = 0;
        bool flag = false;
        
        for(int i = s.size()-1; i >= 0; i--) {
            if(s[i] == ' ' && flag)
                break;
            if(flag)
                len++;
            else if(s[i] != ' ') {
                len++;
                flag = true;
            }
        }
        
        return len;
    }
};

File: LeetCode/length_of_last_word.go
func lengthOfLastWord(s string) int {
    ans, spaces := 0, false

    for i := len(s)-1; i >= 0; i-- {
        if s[i] == ' ' {
            spaces = true
        } else {
            spaces = false
        }

        if spaces && ans > 0 {
            break
        }

        if !spaces {
            ans++
        }
    }

    return ans
}
